### PR TITLE
Translate Intel FPGA force_pow2_depth memory attribute

### PR DESCRIPTION
**Pull request by mlychkov**
https://github.com/KhronosGroup/SPIRV-Registry/blob/master/extensions/INTEL/SPV_INTEL_fpga_memory_attributes.asciidoc

Signed-off-by: Mikhail Lychkov <mikhail.lychkov@intel.com>

### CODE REVIEW

**Comment by mlychkov** [comment]
@MrSidims @AGindinson @AlexeySotkin Please review

File: test/IntelFPGAMemoryAttributes.ll [hunk 1]
```diff
@@ -150,6 +150,25 @@
 ;   s.field = 0;
 ; }
 ;
+; void force_pow2_depth_attr() {
+;   [[intelfpga::force_pow2_depth(0)]] int fp2d_var;
+;
+;   [[intelfpga::force_pow2_depth(1)]] struct fp2d_st {
```
Comment (AGindinson) [inline test/IntelFPGAMemoryAttributes.ll]: Could we have a test-case for `force_pow2_depth(0)` on a `struct`?

File: test/IntelFPGAMemoryAttributesForStruct.ll [hunk 2]
```diff
@@ -556,6 +583,38 @@ entry:
   ret void
 }
 
+; Function Attrs: norecurse nounwind
+define spir_func void @_Z27field_force_pow2_depth_attrv() #3 {
+entry:
+  %s = alloca %struct.force_pow2_depth_st, align 4
+  %0 = bitcast %struct.force_pow2_depth_st* %s to i8*
+  call void @llvm.lifetime.start.p0i8(i64 4, i8* %0) #5
+  ; CHECK-LLVM: %[[FLD_FP2_SCT:.*]] = getelementptr inbounds %struct.force_pow2_depth_st, %struct.force_pow2_depth_st* %{{[a-zA-Z0-9]+}}, i32 0, i32 0
```
Comment (AGindinson) [inline test/IntelFPGAMemoryAttributesForStruct.ll]: IMO, the mapping on a particular global (line 280-281) had best be checked. The globals could be pre-extracted by `CHECK-DAG`s, I believe.

File: test/IntelFPGAMemoryAttributes.ll (hunk 1 above)
Comment (mlychkov) [reply test/IntelFPGAMemoryAttributes.ll]: We may have it, but for what reason? From translator's point of view processing of attribute with any argument value is not depend on a type of variable.

Comment (AGindinson) [reply test/IntelFPGAMemoryAttributes.ll]: Fair enough - just my morning safe-guard mood speaking :)

**Pull request merged**